mpp-ios/ContentView.swift

```diff
@@ -7,16 +7,75 @@
 
 import SwiftUI
 
+struct Station: Identifiable, Hashable {
+    let name: String
+    let id: String
+}
+
 struct ContentView: View {
+    @State private var showingAlert = false
+    
+    var stations = [
+        Station(name: "London Kings Cross", id: "KGX"),
+        Station(name: "Edinburgh Waverley", id: "EDB"),
+        Station(name: "Newcastle", id: "NCL"),
+        Station(name: "York", id: "YRK"),
+        Station(name: "Leeds", id: "LDS")
+    ]
+    @State private var departStation: Station? = nil
+    @State private var arrivalStation: Station? = nil
+    
     var body: some View {
         VStack {
-            Image(systemName: "globe")
-                .imageScale(.large)
-                .foregroundStyle(.tint)
-            Text("Hello, world!")
+            Text("Live Train Times")
+                .font(.title)
+                .fontWeight(.bold)
+                .padding(.all)
+            
+            HStack {
+                Picker("Select departure station", selection: $departStation) {
+                    Text("Select departure...").tag(Optional<Station>(nil))
+                    ForEach(stations) { station in
+                        Text(station.name).tag(station as Station?)
+                    }
+                }
+                .frame(width: 150)
+                Picker("Select arrival station", selection: $arrivalStation) {
+                    Text("Select arrival...").tag(Optional<Station>(nil))
+                    ForEach(stations) { station in
+                        Text(station.name).tag(station as Station?)
+                    }
+                }
+                .frame(width: 150)
+            }
+            Button("Submit") {
```

Comment:
Similar to the pickers this could be its own method
```
    private var submitButton: some View {
        Button("Submit") {
            if departStation == nil || arrivalStation == nil {
                showingAlert = true
            } else {
                let urlString = getUrl(departStation!, arrivalStation!)
                if let url = URL(string: urlString), UIApplication.shared.canOpenURL(url) {
                    UIApplication.shared.open(url)
                }
            }
        }
        .alert("Select departure and arrival stations", isPresented: $showingAlert) {
            Button("OK", role: .cancel) { }
        }
        .padding()
        .font(.title3)
    }
```

Which you'd then call by just using `submitButton`